pysph/base/opencl.py

```diff
@@ -35,6 +35,47 @@ def set_queue(q):
     _queue = q
 
 
+class DeviceArray(cl.array.Array):
+    def __init__(self, queue, dtype, n=0):
+        self.queue = queue
+        self.length = n
+        if n == 0:
+            n = 16
+        self.alloc = n
+        self.dtype = dtype
+        self._array = cl.array.empty(queue, n, dtype)
+
+    @property
+    def data(self):
+        return self._array[:self.length]
+
+    def resize(self, size):
+        self.reserve(size)
+        self.length = size
+
+    def reserve(self, size):
+        if size > self.alloc:
+            self._array = cl.array.empty(self.queue, size, self.dtype)
+            self.alloc = size
+
+    def get_array(self):
+        return self._array
+
+    def set_array(self, array):
```

Comment:
I don't really like the name too much as it isn't really clear what it is doing.  It isn't evident that it is really replacing the underlying data!  Usually set suggests that you are simply setting the values.  Is this really needed?  If you wanted to set a new array, you might as well have array or data as a kwarg to `__init__` making it  obvious and eliminate the need for this method

Reply:
On second thought, I can see the use of this when someone has a reference to this object, so this can stay.